webapp/utils/channel_utils.jsx

```diff
@@ -190,40 +190,44 @@ export function showManagementOptions(channel, isAdmin, isSystemAdmin, isChannel
     return true;
 }
 
-export function showDeleteOption(channel, isAdmin, isSystemAdmin, isChannelAdmin, userCount) {
+export function showDeleteOptionForCurrentUser(channel) {
+    const isTeamAdmin = TeamStore.isTeamAdminForCurrentTeam();
+    const isSystemAdmin = UserStore.isSystemAdminForCurrentUser();
+    const isChannelAdmin = ChannelStore.isChannelAdminForCurrentChannel();
+
```

Comment:
I'm not really sure about this change. The `TeamStore` and `UserStore` singletons are hangovers from Flux that we are working to remove from the code, with the relevant parts of the Redux store now being injected into components through props.

If you want to tidy this up, I think it would make more sense to make this set of functions consistently receive `isSystemAdmin`, `isTeamAdmin` and `isChannelAdmin` parameters, and do away with the rather confusing legacy `isAdmin` parameter that some of them use.